src/co_tools/co_fastq.py

```diff
@@ -0,0 +1,156 @@
+                    break
+            else:
+                prefix_dict[prefix] = [path]
+        else:
+            LOGGER.warning(f"No prefix determined for {path}")
+    if not prefix_dict:
+        LOGGER.error(f"No files found in {dir_path}")
+        return 0
+    if not this_prefix:
+        LOGGER.error(f"fastq files in {dir_path} not properly organized")
+        return 0
+    for path in prefix_dict[this_prefix]:
+        if get_read_direction(path) == "1":
+            fwd = path
+        elif get_read_direction(path) == "2":
+            rev = path
+    if fwd and rev:
+        LOGGER.info(f"returning {fwd},{rev}")
+        return f"{fwd},{rev}"
+    else:
+        LOGGER.error(
+            f"Could not find complementary pair of fastq files in {dir_path}"
+        )
+        return 0
+
+
+def get_fwd_fastqs(dir: str = "../data"):
+    if some_fastq := (
+        subprocess.check_output(["find", "-L", dir, "-name", "*.fastq.gz"])
+        .decode("utf-8")
+        .strip()
+    ):
+        if "\n" in some_fastq:
+            some_fastq = some_fastq.split("\n")[0]
+        LOGGER.debug(f"some_fastq: {some_fastq}")
+        pattern = get_read_pattern(some_fastq)
+        LOGGER.debug(f"pattern: {pattern}")
+        subprocess_files = (
+        subprocess.check_output(["find", "-L", dir, "-name", f"*{pattern}"])
+        .decode("utf-8")
+        .strip()
+    )
+        LOGGER.debug(f"subprocess_files:\n{subprocess_files}")
+        subprocess_files = subprocess_files.split("\n")
+        subprocess_files.sort()
+        LOGGER.debug(f"sorted subprocess_files:\n{subprocess_files}")
+        files = ""
+        for elem in files:
+            files += f"{elem}\n"
+        files = files.strip()
+        LOGGER.debug(f"returning files:\n{files}")
+        return files
+    else:
+        LOGGER.error(f"There are no fastq.gz files in the {dir} directory")
+        return 0
+
+
+def get_read_direction(filepath: str):
+    filename = filepath.split("/")[-1]
+    LOGGER.debug(f"filename: {filename}")
+    if "_" not in filename:
+        LOGGER.warning(
+            "You might be trying to use a single end reads file as a paired"
+            + f" end reads file. Current input: {filepath}"
+        )
+        return 0
+    return "1" if "1" in filename.split("_")[-1].split(".")[0] else "2"
+
+
+def get_read_pattern(filename: str, direction: str = "1"):
+    if "_" not in filename and "/" in filename:
+        LOGGER.warning(
+            f"{filename} might be a single end reads file. The pattern being returned"
+            + " is the entire filename"
+        )
+        return filename.split("/")[-1]
+    direction_complement = "2" if direction == "1" else "1"
+    pattern = filename.split("_")[-1]
+    LOGGER.debug(f"pattern: {pattern}")
+    return (
+        pattern
+        if direction in pattern
+        else pattern.replace(direction_complement, direction)
+    )
+
+
+def get_prefix(filename: str, split_position: str = "-1"):
```

Comment:
This is something I've thought for a while, but we should allow this to take a static suffix in cases where say you want to include the lane number.